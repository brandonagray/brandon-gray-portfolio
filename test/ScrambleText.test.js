import React from 'react';
import ScrambleText from '../src/components/ScrambleText';

describe('ScrambleText component', () => {
  jest.useFakeTimers();
  const fakeText = 'This is a test';

  it('should render the scrambled text', () => {
    const text = mount(
      <ScrambleText
        text={fakeText}
        scramble="!<>-_\\/[]{}—=+*^?#_abiwxevpi"
        options={{ duration: 100, speed: 25 }}
      />
    );
    expect(text).toMatchSnapshot();
  });

  it('should scramble text periodically', () => {
    const text = mount(
      <ScrambleText
        text={fakeText}
        scramble="!<>-_\\/[]{}—=+*^?#_abiwxevpi"
        options={{ duration: 100, speed: 25 }}
      />
    );

    let curText = text.state().curText;

    jest.advanceTimersByTime(150);
    expect(text.state().curText).not.toBe(curText);
    curText = text.state().curText;

    jest.advanceTimersByTime(300);
    expect(text.state().curText).not.toBe(curText);
    curText = text.state().curText;

    jest.advanceTimersByTime(5000);
    expect(text.state().curText).not.toBe(curText);
    expect(text.state().curText).toEqual(fakeText);

    jest.runOnlyPendingTimers();
  });

  it('should clear timer on unmount', () => {
    const text = mount(
      <ScrambleText
        text={fakeText}
        scramble="!<>-_\\/[]{}—=+*^?#_abiwxevpi"
        options={{ duration: 100, speed: 25 }}
      />
    );
    text.unmount();
    expect(clearTimeout).toHaveBeenCalledTimes(3);
  });

  it('should boop on click', () => {
    const text = mount(
      <ScrambleText
        text={fakeText}
        scramble="!<>-_\\/[]{}—=+*^?#_abiwxevpi"
        options={{ duration: 100, speed: 25 }}
      />
    );
    text.simulate('click');
    jest.advanceTimersByTime(1000);
  });
});
