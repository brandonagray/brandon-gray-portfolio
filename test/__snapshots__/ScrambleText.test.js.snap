// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrambleText component should render the scrambled text 1`] = `
.c0 {
  display: inline-block;
  -webkit-animation: none;
  animation: none;
  cursor: default;
}

.c0.boop {
  -webkit-animation: pop 0.3s ease-in-out 1;
  animation: pop 0.3s ease-in-out 1;
}

<ScrambleText
  options={
    Object {
      "duration": 100,
      "speed": 25,
    }
  }
  scramble="!<>-_\\\\\\\\/[]{}â€”=+*^?#_abiwxevpi"
  text="This is a test"
>
  <styled.span
    className={null}
    onAnimationEnd={[Function]}
    onClick={[Function]}
  >
    <span
      className="c0"
      onAnimationEnd={[Function]}
      onClick={[Function]}
    >
       
    </span>
  </styled.span>
</ScrambleText>
`;
